lib/Driver/ToolChains.cpp

```diff
@@ -1286,6 +1300,24 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   if (!Linker.empty()) {
     Arguments.push_back(context.Args.MakeArgString("-fuse-ld=" + Linker));
   }
+  
+  // Configure the toolchain
+  // By default, use the system clang++ to link.
+  const char * Clang = "clang++";
```

Comment:
@compnerd you would still be able to give a particular target-clang by putting it in your tools-directory (or symlinking it to there). That's one of the features of this PR - it lets you override the system 'clang++'.